pkg/cmd/admin/prune/images.go

```diff
@@ -198,6 +177,128 @@ func NewCmdPruneImages(f *clientcmd.Factory, parentName, name string, out io.Wri
 	cmd.Flags().DurationVar(&cfg.KeepYoungerThan, "keep-younger-than", cfg.KeepYoungerThan, "Specify the minimum age of a build for it to be considered a candidate for pruning.")
 	cmd.Flags().IntVar(&cfg.TagRevisionsToKeep, "keep-tag-revisions", cfg.TagRevisionsToKeep, "Specify the number of image revisions for a tag in an image stream that will be preserved.")
 	cmd.Flags().StringVar(&cfg.CABundle, "certificate-authority", cfg.CABundle, "The path to a certificate authority bundle to use when communicating with the OpenShift-managed registries. Defaults to the certificate authority data from the current user's config file.")
+	cmd.Flags().StringVar(&cfg.ServiceAccountName, "service-account", cfg.ServiceAccountName, "If specified, use this service account to perform all pruning operations.")
 
 	return cmd
 }
+
+func getClients(f *clientcmd.Factory, cfg *pruneImagesConfig) (osClient *client.Client, kClient *kclient.Client, registryClient *http.Client, err error) {
+	clientConfig, err := f.OpenShiftClientConfig.ClientConfig()
+	if err != nil {
+		return
+	}
+
+	var token string
+
+	switch {
+	case len(cfg.ServiceAccountName) > 0:
+		var ns string
+		ns, err = f.DefaultNamespace()
```

Comment:
If I do that, and you don't pass namespace/name, I'll still need to call f.DefaultNamespace to be able to do kClient.Secrets(ns), right?